pkg/user/service.go

```diff
@@ -205,3 +210,68 @@ func (s service) Update(id uint, email, password string) (*model.User, error) {
 
 	return s.repository.update(user)
 }
+
+func (s service) sendResetPasswordEmail(user *model.User) error {
+	_, err := s.repository.findByEmail(user.Email)
+	if err != nil {
+		if errdef.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	m := mail.NewMessage()
+	m.SetHeader("From", "DHIS2 Instance Manager <no-reply@dhis2.org>")
+	m.SetHeader("To", user.Email)
+	m.SetHeader("Subject", "Reset your IM password")
+	link := fmt.Sprintf("%s/reset-password/%s", s.config.UIHostname, user.PasswordToken.String)
+	body := fmt.Sprintf("Hello, please click the link below to reset your password.<br/>%s", link)
+	m.SetBody("text/html", body)
+	return s.dailer.DialAndSend(m)
+}
+
+func (s service) RequestPasswordReset(email string) error {
+	user, err := s.repository.findByEmail(email)
+
+	if err != nil {
+		return nil
```

Comment:
Do we really want to silently suppress all errors here?

Reply:
I think we want to do something like [this](https://github.com/dhis2-sre/im-manager/pull/730/files#diff-5a3d7d2e1ba1813b40192dfd630c1919e3fcd8bc1436560e793ce7af2bd98068R216).